src/Tools/dotnet-monitor/Extensibility/OutputParser.cs

```diff
@@ -0,0 +1,75 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Extensions.Logging;
+using System;
+using System.Diagnostics;
+using System.Text.Json;
+using System.Threading.Tasks;
+
+namespace Microsoft.Diagnostics.Tools.Monitor.Extensibility
+{
+    internal class OutputParser<TResult> : IDisposable where TResult : class
```

Comment:
Much of this could be replaced with EventTaskSource: https://github.com/dotnet/diagnostics/blob/main/src/Microsoft.Diagnostics.Monitoring.EventPipe/EventTaskSource.cs

Something like:
```cs
EgressArtifactResult result;
using EventTaskSource<DataReceivedEventHandler> resultSource = new(
    complete => (s, e) => { ... }, // This does the deserialization and invokes complete when the deserialization successfully finishes
    handler => process.OutputDataReceived += handler,
    handler => process.OutputDataReceived -= handler
)

...

await resultSource.Task;

```

Reply:
The idea is this will hold all parsing logic in the future.